packages/kbn-custom-integrations/src/components/create/form.tsx

```diff
@@ -0,0 +1,209 @@
+        },
+      });
+    },
+    [send]
+  );
+
+  const retry = useCallback(() => {
+    send({ type: 'RETRY' });
+  }, [send]);
+
+  const hasFailed = useSelector(machineRef, hasFailedSelector);
+
+  return (
+    <CreateCustomIntegrationForm
+      integrationName={state?.context.fields.integrationName}
+      datasetName={state?.context.fields.datasets[0].name} // NOTE: Hardcoded for now until we support multiple datasets
+      errors={state.context.errors}
+      updateIntegrationName={updateIntegrationName}
+      updateDatasetName={updateDatasetName}
+      touchedFields={state?.context.touchedFields}
+      hasFailed={hasFailed}
+      onRetry={hasFailed ? retry : undefined}
+    />
+  );
+};
+
+interface FormProps {
+  integrationName: CreateCustomIntegrationOptions['integrationName'];
+  datasetName: Dataset['name'];
+  errors: WithOptionalErrors['errors'];
+  touchedFields: WithTouchedFields['touchedFields'];
+  updateIntegrationName: (integrationName: string) => void;
+  updateDatasetName: (integrationName: string) => void;
+  hasFailed: boolean;
+  onRetry?: () => void;
+}
+
+export const CreateCustomIntegrationForm = ({
+  integrationName,
+  datasetName,
+  errors,
+  touchedFields,
+  updateIntegrationName,
+  updateDatasetName,
+  onRetry,
+}: FormProps) => {
+  return (
+    <>
+      <EuiText color="subdued">
+        <p>
+          {i18n.translate('customIntegrationsPackage.create.configureIntegrationDescription', {
+            defaultMessage: 'Configure integration',
+          })}
+        </p>
+      </EuiText>
+      <EuiSpacer size="l" />
+      <EuiForm fullWidth>
+        <EuiFormRow
+          label={
+            <EuiFlexGroup alignItems="center" gutterSize="xs" responsive={false}>
+              <EuiFlexItem grow={false}>
+                {i18n.translate('customIntegrationsPackage.create.integration.name', {
+                  defaultMessage: 'Integration name',
+                })}
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiIconTip
+                  content={i18n.translate(
+                    'customIntegrationsPackage.create.integration.name.tooltip',
+                    {
+                      defaultMessage:
+                        'Provide a name for the integration that will be created to organise these custom logs.',
+                    }
+                  )}
+                  position="right"
+                />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          }
+          helpText={i18n.translate('customIntegrationsPackage.create.integration.helper', {
+            defaultMessage:
+              "All lowercase, max 100 chars, special characters will be replaced with '_'.",
+          })}
+          isInvalid={hasErrors(errors?.fields?.integrationName) && touchedFields.integrationName}
+          error={errorsList(errors?.fields?.integrationName)}
+        >
+          <EuiFieldText
```

Comment:
**suggestion:** as an extra limit to prevent user writing more than necessary, you could add the `max` attribute to this field, which will block the insertion of more chars than those allowed. Same applies for the dataset name field.